src/api/04-enhancements/12-background-fetch-api.md

```diff
@@ -0,0 +1,166 @@
+
+The [Background Fetch API] allows web applications to offload a download operation to the browser. The browser exposes some UI to indicate that a download is in progress. When all files are downloaded, it exposes the downloaded data to your service worker.
+
+[Background Fetch API] allows you to batch many files into a single download operation. This is especially useful for downloading media, especially for downloading streaming formats like MPEG DASH that often serve a single media item split into tens or even hundreds of chunks, each in its own file.
+
+## Initiate the download
+
+Your application needs to have a registered service worker in order to use the [Background Fetch API]. You will also need to make sure the API is supported in the user's browser before you use it.
+
+```js
+if (!('BackgroundFetchManager' in self)) {
+  // Use the Fetch API instead to download the files.
+}
+```
+
+Assuming the current browser supports the Background Fetch API, you should first obtain a `ServiceWorkerRegistration` instance and then use its `backgroundFetch` property to initialize the background fetch operation.
+
+```js
+navigator.serviceWorker.ready.then(async (swReg) => {
+  const bgFetch = await swReg.backgroundFetch.fetch(
+    // Provide your own meaningful ID.
+    'my-video-id',
+
+    // Array of file URLs to be downloaded.
+    [
+      '/static/video-chunk-1.mp4',
+      '/static/video-chunk-2.mp4',
+      '/static/video-chunk-3.mp4',
+      '/static/video-chunk-4.mp4',
+      '/static/video-thumbnail.jpg',
+    ],
+
+    // Browsers may use these options
+    // to provide users with more information
+    // about the object being downloaded.
+    //
+    // Note: All options are optional.
+    {
+      title: 'My Video: Chunk Funk',
+      icons: [{
+        sizes: '300x300',
+        src: '/video-icon-300x300.png',
+        type: 'image/png',
+      }],
+
+      // Total size of the downloaded
+      // resources in bytes.
+      downloadTotal: 60 * 1024 * 1024,
+    }
+  );
+});
+```
+
+## Access downloaded data
+
+Once the background fetch is initialized, the browser is going to expose some kind of progress indication to the user. The download will now continue even if the user closes the application or navigates away.
+
+After all files finish downloading, the browser is going to trigger a `backgroundfetchsuccess` event in your service worker context. This gives you an opportunity to access the downloaded data.
+
+```js
+// In the service worker ↓
+
+addEventListener('backgroundfetchsuccess', (event) => {
+  // The `BackgroundFetchRegistration` instance.
+  const bgFetch = event.registration;
+
+  event.waitUntil(async function() {
+    // Returns an array of `BackgroundFetchRecord` objects.
+    const records = await bgFetch.matchAll();
+
+    const promises = records.map(async (record) => {
+      const response = await record.responseReady;
+
+      // Maybe store the `Response` object using
```

Comment:
In this case, we'd want to store those `Response` objects.
Maybe `TODO: Store the response object using the Cache or IndexedDB API`

Reply:
Updated in 281290836b44cdc445dea0ef7410d1082e9d8506.